### PR TITLE
Added game logic branch: milestone3

### PR DESCRIPTION
**Pull request by kimothokamau**
This pull request adds the game logic to replace the mock moves.
The milestone revolves around Object Oriented Programming concept. With a focus on how objects, classes, and methods interact with each other.

### CODE REVIEW

File: lib/game_logic.rb
Comment (cristianCeamatu) [inline lib/game_logic.rb]: - [ ] No `puts/print` and `gets.chomp` outside `main.rb` file. ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/OOP/milestone3/only_main_print.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/59565018/)) In order to fix this requirement, you can return a string from the `display_board` method and use `puts display_board` in the `main.rb` file.

**Review by cristianCeamatu — CHANGES_REQUESTED** [review]
## ♻️ Changes required
#### Congratulations for your efforts! The logic was easy to follow and understand and your project works perfectly! Very small methods that do one thing, resulting in a great DRY principle.
However, there are some small changes to be done in order to get this project approved:

### Mandatory
- [ ] The project uses at least two classes (Usually it uses three: Game, Board, and Player). The code of the project should be logically divided between classes. ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/OOP/milestone3/classes.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/59565018/)) . You created only one class for all your logic, this is a great opportunity to improve your OOP skills, try to think about what responsibility should a `Player`, `Game`, or `Board` class have, then you could divide the methods between them.

### Suggestions
1. [Optional] Add `sleep` and colors to your game, this will totally transform your game and make it feel like it is not a simple CLI game. Links: [colors (the second answer is without using any gem)
](https://stackoverflow.com/questions/1489183/colorized-ruby-output-to-the-terminal), [sleep](https://stackoverflow.com/questions/1329967/tell-ruby-program-to-wait-some-amount-of-time) and a working example: [example](https://repl.it/@cristianCeamatu/microverse-ruby-tic-tac-toe)

### 👏  For already including a decent Readme in this milestone

## Please also check my inline comments and do the required changes then submit for another review

Don't hesitate to contact me on Slack if needed, I would be happy to join your call and help.

Best regards,
Cristian Ceamatu
Slack: Cristian Viorel Ceamatu

**Review by MahmoudBakr23 — APPROVED** [review]
# **Approved ✔✔**

<hr>

Hello Team. Good job you've done so far in this milestone and in the requested changes!

<hr>

## [Better to do]

- Modify your readme file to make it ```ruby ./bin/main.rb``` or ```bin/main.rb``` instead of ```ruby main.rb```

## [NICE TO HAVE]

- It'd be great if you deploy your app on a live demo link i.e. repl.it for easier access

<hr>

You can now merge and move on to the next milestone, and great job making such an excellent README.md file in earlier stages. 👍👍

Good luck and happy coding 👨‍💻💻

**Pull request merged**
